### PR TITLE
#162380294 create static signup page

### PR DESCRIPTION
**Pull request by sulenchy**
### What is it this PR doing

Create static signup page

#### Description of Task to be completed?

- create the small screen view
- create the big screen view
- add styles
- write snapshot tests for the component

#### How should this be manually tested?

- clone this repo
- branch out to feature/162380294/static-signup-page branch
- run npm install
- run npm run start:dev

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?

[#162380294](https://www.pivotaltracker.com/story/show/162380294)

#### Screenshots (if appropriate)

screen shot 2018-12-08 at 7 20 38 pm
screen shot 2018-12-08 at 7 21 02 pm

#### Questions:
N/A

#### Link to review App

[Review App](https://valinor-ah-frontend-stag-pr-15.herokuapp.com/signup)

### CODE REVIEW

**Comment by augustineezinwa** [comment]
@sulenchy Your markup for the desktop view seems not to be centralized. I suggest you add more paddings or margin beneath to get the markup centralized. Refer to the desktop figma mockup for more details.
Also for the signup on the mobile view. Your signup button is to be touching the bottom of a  phone screen. Consider adding padding to that button.

File: public/assets/styles/index.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
+@import url(https://fonts.googleapis.com/css?family=Open+Sans|Zilla+Slab);
+
+@import '../../../public/assets/styles/signup';
+
+$bluecolor: #b7daf72d;
+$color-bg: #f4f1f4;
+$color-bg-light-blue:#007fff;
```
Comment (KvNGCzA) [inline public/assets/styles/index.scss]: all variables should be in a variables.scss file

File: public/assets/styles/signup.scss [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+$bluecolor: #b7daf72d;
+$color-bg: #f4f1f4;
+$color-bg-light-blue:#007fff;
+$color-black: #000000;
+$color-blue: #1a1aff;
+$color-facebook: #3b5998;
+$color-twitter: #1da1f2;
+$nav-item-black: #3a3a3a;
+$nav-red: #ff0000;
```
Comment (KvNGCzA) [inline public/assets/styles/signup.scss]: all variables should be in a variables.scss file

File: public/assets/styles/signup.scss [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    }
+  
+    .blue-section{
+      background-color: $color-bg-light-blue;
+      border-radius: 10px 0 0 10px;
+      color: $color-white;
+      height: $height-section;
+    }
+  
+    @mixin btn-rounded-basic(){
```
Comment (KvNGCzA) [inline public/assets/styles/signup.scss]: all mixins should be in a mixins.scss file

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

**Review by tersoo-atsen — APPROVED** [review]

File: public/assets/styles/index.scss [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  nav .nav-item:hover {
+    color: $nav-red;
+  }
+
+}
+
+h1 {
+  text-align: center;
+}
```
Comment (KvNGCzA) [inline public/assets/styles/index.scss]: Line 8 to line 38 are not needed anymore. They are already declared below.

File: public/assets/styles/mixins.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
+@mixin btn-rounded-basic(){
+    border-radius: 30px;
```
Comment (KvNGCzA) [inline public/assets/styles/mixins.scss]: I think this should be `em` like line 8 below

File: public/assets/styles/mixins.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+@mixin btn-social-basic {
+    border-radius: .3em;
+    color: $text-white;
+    margin: .2em;
+    padding: .7em;
+    width: 12.5em;
+    border: 0;
+}
+
+@mixin outerBoxShadow($v,$w, $x, $y, $z){
```
Comment (KvNGCzA) [inline public/assets/styles/mixins.scss]: we have renamed this mixin outer-box-shadow. Please do the same.

File: public/assets/styles/mixins.scss [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    -moz-box-shadow: $v $w $x $y $z;
+    -webkit-box-shadow: $v $w $x $y $z;
+    -ms-box-shadow: $v $w $x $y $z;
```
Comment (KvNGCzA) [inline public/assets/styles/mixins.scss]: please remove this line. We wont need it.

File: public/assets/styles/signup.scss [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+      color: $text-white;
+    }
+    
+    .btn-rounded-blue{
+      @include btn-rounded-basic();
+      border: 1px solid $color-bg-light-blue;
+      color: $color-bg-light-blue;
+      margin-top: 1.5em;
+    }
+    $line-color: #ced4da;
```
Comment (KvNGCzA) [inline public/assets/styles/signup.scss]: This variable is out of place. It should be in the variables.scss file

File: public/assets/styles/signup.scss [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+
+  .blue-section .header{
+    margin-top: 130px;
+  }
+  
+  }
+  
+  @media screen and (max-width: 850px) {
+    /* phones */
+    .main-container .site-content {
```
Comment (KvNGCzA) [inline public/assets/styles/signup.scss]: all styles that are going to affect general div's that everyone will have should be in the index.scss file so we dont have to search through all scss files if something goes wrong.

File: public/assets/styles/signup.scss [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+      border: 0;
+      border-bottom: 1px solid $line-color;
+      border-radius: 0;
+    }
+
+    .email-input {
+      @extend %input-style;
+      background-color: transparent;
+
+      &::placeholder {
```
Comment (KvNGCzA) [inline public/assets/styles/signup.scss]: use this mixin for placeholders
```
 @mixin placeholder($color){
  &::-webkit-input-placeholder{
    color: $color;
  }
  &::-moz-placeholder{
    color: $color;
  }
  &:-ms-input-placeholder{
    color: $color;
  }
  &:-moz-placeholder{
    color: $color;
  }
  &::placeholder{
    color: $color;
  }
}
```

File: public/assets/styles/index.scss [hunk 11]
```diff
@@ -0,0 +1,79 @@
+@import url(https://fonts.googleapis.com/css?family=Zilla+Slab:300,300i,400,500,600,700%7COpen+Sans:300,400,600,700,800);
+
+
+@import './variables.scss';
+@import './mixins.scss';
+@import '../../../public/assets/styles/signup';
```
Comment (KvNGCzA) [inline public/assets/styles/index.scss]: this path should be `./signup`. please take out the scss extension on lines 4 and 5.

**Review by KvNGCzA — COMMENTED** [review]
Please make all `px` em.

**Review by KvNGCzA — APPROVED** [review]

**Review by augustineezinwa — APPROVED** [review]
LGTM

File: src/components/signup/SignUpPage.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    return (
+      <Fragment>
+        <Header />
+        <div className="site-content">
+          <div className="container section">
+            <div className="row justify-content-md-center">
+              <div className="col-sm-12 col-md-4 blue-section">
+                <div className="header">
+                  <div className="box-intro">
+                    <h1 id="welcome">Welcome to Author's Haven!</h1>
```
Comment (mbilesanmi) [inline src/components/signup/SignUpPage.js]:
```suggestion
                    <h1 id="welcome">Welcome to Author&apos;s Haven!</h1>
```

File: src/components/signup/SignUpPage.js [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+                  </button>
+                </div>
+                <div className="signup-form">
+                  <form className="form-signup">
+                    <span className="text-sign-intro">sign up with your email</span>
+                    <div className="input-group mb-3">
+                      <div className="input-group-prepend">
+                        <span className="input-group-text transparent email-addon" id="basic-addon1"><img src={userIcon} alt=" email icon" /></span>
+                      </div>
+                      <input type="email" className="form-control email-input transparent" placeholder="FULL NAME" aria-label="Username" aria-describedby="basic-addon1" />
```
Comment (mbilesanmi) [inline src/components/signup/SignUpPage.js]: Why is this `type="email"`?

**Review by mbilesanmi — CHANGES_REQUESTED** [review]

File: src/components/signup/SignUpPage.js (hunk 12 above)
Comment (sulenchy) [reply src/components/signup/SignUpPage.js]: Feedback duly noted and implemented

File: src/components/signup/SignUpPage.js (hunk 13 above)
Comment (sulenchy) [reply src/components/signup/SignUpPage.js]: Feedback duly noted and implemented

**Review by mbilesanmi — APPROVED** [review]

**Pull request merged**